renter/download.go

```diff
@@ -340,3 +349,13 @@ func convertToReaderSlice(files []*os.File) []io.Reader {
 	}
 	return res
 }
+
+
+func findVersion(file *core.File, versionNum int) *core.Version {
+	for i := 0; i < len(file.Versions); i++ {
```

Comment:
You can fetch versions directly from the metaserver using the endpoint `/files/{id}/versions/{id}` or something to that effect (it's in the metaclient). Could be useful if you want the same behavior I used for getting shared files (check first in cache, then check metaserver).